uktrade/data-hub-frontend
# Add company one list edit form

## Description of change

Adds a form that allows permitted users to update One List information, for example:
- Adding an existing company to a One List
- Changing a company One List Tier, Global Account Manager or Core Team
- Removing a company from One List
- Removing Core Team members

## Test instructions
If necessary, go to Django admin to make sure you have the 'Change One List Tier and Global Account Manager' and the 'Change Company' company permissions

If viewing the site as a user with the permission:
- Visit Companies > select a Company > View Full Business Details
- Scroll to the bottom of the page and click on the 'Edit One List Information' button
- You should see the first stage of the form, with one list information pre-filled (if any)
- If you click 'Not on the One List', the 'Continue' button will change to 'Submit' and the form should submit, redirecting you to the page Business Details on success
- If you click on another tier, click on 'Continue' and progress through the second stage of the form
- On clicking 'Submit', the form should submit and you will be directed to the Business Details page on success

If viewing the site as a user without the right permissions:
- Visit Companies > select a Company > View Full Business Details
- Scroll down the page and you should not see the 'Edit One List Information' button

## Screenshots

![page_1](https://user-images.githubusercontent.com/23265724/83274566-e6247680-a1c5-11ea-94be-1954340cafdc.gif)
<img width="704" alt="page 2" src="https://user-images.githubusercontent.com/23265724/83274561-e4f34980-a1c5-11ea-86a4-d6f90e24c49d.png">
<img width="710" alt="page 3" src="https://user-images.githubusercontent.com/23265724/83274558-e3c21c80-a1c5-11ea-8c77-6b431b38be5e.png">
<img width="703" alt="page 4" src="https://user-images.githubusercontent.com/23265724/83274546-e0c72c00-a1c5-11ea-9d01-04c17b56993c.png">


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
